### PR TITLE
benchmark: add new versions 1.9.0 and 1.8.5

### PR DESCRIPTION
**Pull request by toge**
### Summary
Changes to recipe:  **benchmark/1.8.5**

#### Motivation
benchmark requires C++14 since 1.8.5.

#### Details
https://github.com/google/benchmark/compare/v1.8.4...v1.8.5#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20a

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Review by AbrilRBS — COMMENTED** [review]
The recipe might be building with C++14 regardless of `compiler.cppstd`, we might need to patch out [google/benchmark@`v1.8.4...v1.8.5`#diff-1e7de1ae2d](https://github.com/google/benchmark/compare/v1.8.4...v1.8.5#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR143)

File: recipes/benchmark/all/conanfile.py [hunk 1]
```diff
@@ -35,6 +35,20 @@ class BenchmarkConan(ConanFile):
         "enable_libpfm": False,
     }
 
+    @property
+    def _min_cppstd(self):
+        return 14
```
Comment (valgur) [inline recipes/benchmark/all/conanfile.py:40]:
```suggestion
        return 14 if Version(self.version) >= "1.8.5" else 11
```
Shouldn't C++11 be set for the older versions?

**Review by jwillikers — APPROVED** [review]

**Review by ErniGH — APPROVED** [review]
Thanks! 👍

**Review by AbrilRBS — APPROVED** [review]

**Review by AbrilRBS — APPROVED** [review]
Sorry for the noise, didn't realize that it was 1.7.1 the version that started hardcoding the standard :)

**Review by jwillikers — APPROVED** [review]

**Pull request merged**